path_collector.go

```diff
@@ -230,3 +241,96 @@ func ExcludeWithGitignore(gitignorePath string, paths []string) ([]string, error
 	logger.Debug(logger.DebugCodePaths, "paths to format: %s", pathsToFormat)
 	return pathsToFormat, nil
 }
+
+const DefaultPatternFile = "yamlfmt.patterns"
+
+// PatternFile determines which files to format and which to ignore based on a pattern file in gitignore(5) syntax.
+type PatternFile struct {
```

Comment:
Can we rename this to `PatternFileCollector` to make clear that it is an implementation of `PathCollector` like the other types in this file?

Reply:
✅ done